vector-im/element-android
# A11y fixes

## Type of change


## Content


This PR improve the behavior of the application regarding accessibility.
- It fixes some bugs observed when navigating with the keyboard (using arrows and tab).
- It also improves behavior when the screen reader is active, by given accessibility focus on Alert displayed on top of the screen, such as verification request, or incoming phone call.
- On the Alert it adds a custom action to let visually impaired users discover how to close the alert (there is no close button, user has to laterally swipe the alert).
- Content description of avatar has been updated to give more info when the screen reader is one.

## Motivation and context



## Screenshots / GIFs





## Tests



- Navigate into the app using the keyboard. For instance, in the room search screen, it was not possible to navigate from the toolbar to the room list and extra action at the bottom for instance. Now it's possible
Note: In the room list, you can navigate to the top of the screen, and to enter in the toolbar content, user must press on "Tab" (twice)
- Enable screen reader from the accessibility setting of the phone and start a verification flow. Screen reader should focus and read the first item on each new step, to let user know that a new step is displayed. There are some glitch here, sometimes the last item is focused.

*Note*: all those changes should not have any impact if no keyboard is connected to the device, nor if no screen reader is enabled.

## Tested devices

- [x] Physical, using [Vysor](https://www.vysor.io/) as a gateway to control the device with the keyboard of the computer.
- OS version(s):

## Checklist
